ocs_ci/deployment/provider_client/storage_client_deployment.py

```diff
@@ -0,0 +1,338 @@
+            "namespace"
+        ] = namespace_to_create_storage_client
+
+        # Create network policy
+        network_policy_data_yaml = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="network_policy", delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            network_policy_data, network_policy_data_yaml.name
+        )
+        log.info("Creating NetworkPolicy CR")
+        self.ocp_obj.exec_oc_cmd(f"apply -f {network_policy_data_yaml.name}")
+
+    def onboarding_token_generation_from_ui(
+        self,
+    ):
+        """
+        This method generates onboarding token from UI
+
+        Steps:
+        1:- Check private and public keys are available
+        2:- Check Storage-Clients pages available
+
+        Returns:
+        onboarding_token(str): client onboarding token
+
+        """
+        secret_ocp_obj = ocp.OCP(
+            kind=constants.SECRET, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        for secret_name in {
+            constants.ONBOARDING_PRIVATE_KEY,
+            constants.MANAGED_ONBOARDING_SECRET,
+        }:
+            assert secret_ocp_obj.is_exist(
+                resource_name=secret_name
+            ), f"{secret_name} does not exist in {config.ENV_DATA['cluster_namespace']} namespace"
+
+        # verify storage-client page is available
+        validation_ui_obj = ValidationUI()
+        onboarding_token = validation_ui_obj.verify_storage_clients_page()
+        return onboarding_token
+
+    def create_client(
+        self,
+        storage_provider_endpoint,
+        onboarding_token,
+    ):
+        """ """
+        # Pull storage-client yaml data
+        log.info("Pulling storageclient CR data from yaml")
+        storage_client_data = templating.load_yaml(constants.STORAGE_CLIENT_YAML)
+
+        # Set storage provider endpoint
+        log.info(
+            "Updating storage provider endpoint details: %s", storage_provider_endpoint
+        )
+        storage_client_data["spec"][
+            "storageProviderEndpoint"
+        ] = storage_provider_endpoint
+
+        # Set onboarding token
+        log.info("Updating storage provider endpoint details: %s", onboarding_token)
+        storage_client_data["spec"]["onboardingTicket"] = onboarding_token
+        storage_client_data_yaml = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="storage_client", delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            storage_client_data, storage_client_data_yaml.name
+        )
+        log.info("Creating storageclient CR")
+        self.ocp_obj.exec_oc_cmd(f"apply -f {storage_client_data_yaml.name}")
+        # replace_content_in_file(constants.STORAGE_CLIENT_YAML, "PLACEHOLDER1", storage_provider_endpoint)
+        # replace_content_in_file(constants.STORAGE_CLIENT_YAML, "PLACEHOLDER2", onboarding_token)
+        # log.info(f"cluster yaml file: {storage_client_yaml}")
+
+    def test_deployment(self):
```

Comment:
suggest this test to place in libtests folder and marked as a libtest

Reply:
I have added this for time being to test the code and run it locally. It will be updated shortly.